server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java

```diff
@@ -2566,6 +2567,61 @@ protected DiskOfferingVO createDiskOffering(final Long userId, final Long domain
             isCustomized = true;
         }
 
+        if (minIopsPerGb != null || maxIopsPerGb != null) {
+
+            if (!isCustomized) {
+               throw new InvalidParameterValueException("Cannot set Min/Max IOPS/GB for a fixed size disk offering");
+            }
+
+            if ((isCustomizedIops != null && isCustomizedIops) || minIops != null || maxIops != null) {
+                throw new InvalidParameterValueException("Cannot set Min/Max IOPS/GB with either " +
+                        "custom IOPS or fixed IOPS");
+            }
+
+            if (minIopsPerGb != null && maxIopsPerGb != null) {
```

Comment:
Could you extract this block to a specific method? Document it, and write unit tests for it?